### ISSUE TITLE
pyglet 1.4.10 - spureous traceback quiting test scripts

### ISSUE THREAD

**Issue by ccanepa** [los-cocos/cocos#334]
1. all of `test/test_*` ends with some trivial exception if quiting with ESC or in the programatic termination done in `autotest\recon.py`.

The exception does not happen if quiting with the window's close button

problem: this polutes the report generated by recon.py, now all test are marked as having tracebacks.

win7, py3.8, cocos b490c, pyglet 1.4.10

Additionally, between the noise was spotted

```
proxy run cmdline: ['D:\\dev\\cocos2020\\tools\\autotest\\proxy_snapshots.py', 'test_shader.py', 't 0.1, s, q', 'D:\\dev\\cocos2020\\test\\snp']
err: Exception ignored in: <function ShaderProgram.__del__ at 0x0000000006EB5E50>

Traceback (most recent call last):

  File "d:\dev\cocos2020\cocos\shader.py", line 228, in __del__

  File "d:\dev\cocos2020\cocos\shader.py", line 232, in destroy

  File "d:\dev\pyglet-2019\pyglet\gl\lib_wgl.py", line 93, in __call__

ImportError: sys.meta_path is None, Python is likely shutting down

Exception ignored in: <function Shader.__del__ at 0x0000000006EB5550>

Traceback (most recent call last):

  File "d:\dev\cocos2020\cocos\shader.py", line 79, in __del__

  File "d:\dev\cocos2020\cocos\shader.py", line 141, in destroy

  File "d:\dev\pyglet-2019\pyglet\gl\lib_wgl.py", line 93, in __call__

ImportError: sys.meta_path is None, Python is likely shutting down

Error in sys.excepthook:

Original exception was:

``` 

in pure pyglet

 - noisy.py does not present this defect
 - astraea.py shows same behavior than cocos test_* , and also shows the exception when quiting with keypress over the 'quit' menuitem

**Comment by ccanepa** [comment]
opened https://github.com/pyglet/pyglet/issues/142

**Comment by ccanepa** [comment]
The problem does not manifest with py 37

**Comment by ccanepa** [comment]
testing for cocos 0.6.8 done with py37

**Comment by ccanepa** [comment]
fixed in commit cab0b8a by
 - move the termination code outside of on_draw
 - calling director.window.close() instead of pygleet.app.quit()
 - special casing related to test_shader and test_pyglet_vb
